src/coredump/coredump.c

```diff
@@ -322,18 +324,99 @@ static int make_filename(const Context *context, char **ret) {
         return 0;
 }
 
+static int get_cgroup_memory_limit(uint64_t *ret_max_size) {
+        uint64_t unit_memory_current = 0, unit_memory_max = 0, slice_memory_current = 0, slice_memory_max = 0;
+        _cleanup_free_ char *unit = NULL, *unit_path = NULL, *slice = NULL, *slice_path = NULL;
+        _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
+        int r;
+
+        assert(ret_max_size);
+
+        r = sd_pid_get_unit(0, &unit);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get current unit name: %m");
+
+        r = sd_bus_path_encode("/org/freedesktop/systemd1/unit", unit, &unit_path);
+        if (r < 0)
+                return log_error_errno(r, "Failed to encode unit %s DBUS path: %m", unit);
+
+        r = sd_pid_get_slice(0, &slice);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get current slice name: %m");
```

Comment:
hmm, a slice further up might have some limit set too.

I fear we have to move this into pid1. i.e. expose a new dbus property that reports the effective memory limit of a unit, which is determined by any limit set on the unit, and then lowered to any limit enforced on any parent slice unit. and then maybe even cut off by the amount of installed memory.

i.e. this would then be a one-stop solution: if you enquire this property it tells you "this much is how much you have available at most".

Reply:
makes sense, I'll work on that asap - might come back for pointers

Reply:
implemented as a new MemoryAvailable property